starter-code/src/functions-and-arrays.js

```diff
@@ -11,14 +14,73 @@ var words = [
   'crackpot'
 ];
 
+function findLongestWord(arr) {
+  let longestWord = arr[0];
+  for (let i=1; i < arr.length; i++) {
+    if (arr[i].length > longestWord.length) {
+      longestWord = arr[i];
+    }
+  }
+  return longestWord;
+}
+findLongestWord(words);
+
 // Calculating a Sum
 
 var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumArray(arr) {
+  let total = 0;
+    for (i=0; i<arr.length; i++) {
+      total += arr[i];
+    }
+    return total;
+}
+sumArray(numbers);
+
 // Calculate the Average
 
 var numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(arr) {
+  if( arr == false) {
+    return undefined;
+  }
+  let total = 0;
+    for (i=0; i<arr.length; i++) {
+      total += arr[i];
+    }
+    return total / arr.length;
+}
+averageNumbers(numbersAvg);
+
+// Calculate the Average Part Deux
+var words = [
+  'seat',
+  'correspond',
+  'linen',
+  'motif',
+  'hole',
+  'smell',
+  'smart',
+  'chaos',
+  'fuel',
+  'palace'
+];
+
+function averageWordLength(arr) {
+  if( arr == false) {
+    return undefined;
+  }
+  let arrayWordLength = [];
+  for (i=0; i<arr.length; i++) {
+    arrayWordLength.unshift(arr[i].length);
```

Comment:
Nothing wrong with using unshift to add elements to the array, but it's worth nothing that push is more commonly used, and often considered best practice.